### PR TITLE
Add ResponseCompression integration tests

### PR DESCRIPTION
**Pull request by Tratcher**
@JunTaoLuo @BrennanConroy 
I still need to verify the behavior of NGinx, we'll see what Travis says.

### CODE REVIEW

File: test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+                    TargetFramework = runtimeFlavor == RuntimeFlavor.Clr ? "net451" : "netcoreapp1.0",
+                    ApplicationType = applicationType
+                };
+
+                using (var deployer = ApplicationDeployerFactory.Create(deploymentParameters, logger))
+                {
+                    var deploymentResult = deployer.Deploy();
+                    var httpClientHandler = new HttpClientHandler() { AutomaticDecompression = DecompressionMethods.None };
+                    Assert.True(httpClientHandler.SupportsAutomaticDecompression);
+                    var httpClient = new HttpClient(httpClientHandler) { BaseAddress = new Uri(deploymentResult.ApplicationBaseUri) };
```
Comment (BrennanConroy) [inline test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: `using httpClient` please!

File: test/ServerComparison.TestSites/StartupResponseCompression.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            });
+            */
+            app.Run(context =>
+            {
+                context.Response.ContentType = "text/plain";
+                string body;
+                if (context.Request.Path.Value == "/")
+                {
+                    body = "Running";
+
```
Comment (BrennanConroy) [inline test/ServerComparison.TestSites/StartupResponseCompression.cs]: nit: `\r\n`

**Comment by BrennanConroy** [comment]
:shipit:

File: test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+                    await scenario(httpClient, logger);
+                }
+            }
+        }
+
+        private static async Task CheckNoCompressionAsync(HttpClient client, ILogger logger)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, "NoAppCompression");
+            request.Headers.AcceptEncoding.ParseAdd("gzip,deflate");
+            var response = await client.SendAsync(request);
```
Comment (JunTaoLuo) [inline test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: Hmm does this need a RetryHelper?

File: test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+            var request = new HttpRequestMessage(HttpMethod.Get, "Identity");
+            request.Headers.AcceptEncoding.ParseAdd("gzip,deflate");
+            var response = await client.SendAsync(request);
+            var responseText = await response.Content.ReadAsStringAsync();
+            try
+            {
+                Assert.Equal(HelloWorldBody, responseText);
+                Assert.Equal(HelloWorldBody.Length.ToString(), GetContentLength(response));
+                Assert.Equal(1, response.Content.Headers.ContentEncoding.Count);
+                Assert.Equal("identity", response.Content.Headers.ContentEncoding.First());
```
Comment (JunTaoLuo) [inline test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: `.Single()` can replace these two lines.

Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: This gives better error messages when it fails

File: test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs (hunk 3 above)
Comment (Tratcher) [inline test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs]: No, that's already taken care of in the main method.

**Pull request merged**